sources/packages/api/src/route-controllers/education-program-offering/education-program-offering.controller.ts

```diff
@@ -32,7 +46,23 @@ export class EducationProgramOfferingController {
     @Body() payload: SaveEducationProgramOfferingDto,
     @Param("locationId") locationId: number,
     @Param("programId") programId: number,
+    @UserToken() userToken: IInstitutionUserToken,
   ): Promise<number> {
+    const requestedLoc = await this.locationService.getInstitutionLocationById(
+      locationId,
+    );
+    if (
```

Comment:
Code from line 54-58 has been repeated in multiple places. 
Hence this piece of code can be added to a method and called. It will reduce the redundant code.